backend-rust/src/graphql_api.rs

```diff
@@ -3346,6 +3559,56 @@ impl SearchResult {
     }
 }
 
+fn decode_value_with_schema(
+    opt_schema: &Option<Type>,
+    schema_name: &str,
+    value: &Vec<u8>,
+    value_name: &str,
+) -> String {
+    match opt_schema {
+        Some(schema) => {
+            let mut cursor = Cursor::new(&value);
+            match schema.to_json(&mut cursor) {
+                Ok(v) => {
+                    to_string(&v).unwrap_or_else(|e| {
+                        // We don't return an error here since the query is correctly formed and
+                        // the CCDScan backend is working as expected.
+                        // A wrong/missing schema is a mistake by the smart contract
+                        // developer which in general cannot be fixed after the deployment of
+                        // the contract. We display the error message (instead of the decoded
+                        // value) in the block explorer to make the info visible to the smart
+                        // contract developer for debugging purposes here.
+                        format!(
+                            "Failed to deserialize {} with {} schema into string: {:?}",
+                            value_name, schema_name, e
+                        )
+                    })
+                }
+                Err(e) => {
+                    // We don't return an error here since the query is correctly formed and
+                    // the CCDScan backend is working as expected.
+                    // A wrong/missing schema is a mistake by the smart contract
+                    // developer which in general cannot be fixed after the deployment of
+                    // the contract. We display the error message (instead of the decoded
+                    // value) in the block explorer to make the info visible to the smart
+                    // contract developer for debugging purposes here.
+                    format!(
+                        "Failed to deserialize {} with {} schema: {:?}",
+                        value_name,
+                        schema_name,
+                        e.display(true)
+                    )
+                }
+            }
+        }
+        // Note: Shall we use something better than this string if no schema is
+        // available for decoding.
```

Comment:
Maybe we could rather just return a `null` or some kind of missing value to the frontend and the frontend can display something more intelligent than just a string error message. I.e. it could display something graphical. But we're not looking to change the frontend yet so it's maybe okay to stick with this for now?